lib/Dialect/Handshake/HandshakeOps.cpp

```diff
@@ -2261,5 +2261,20 @@ LogicalResult TruncIOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ExtFOp
+//===----------------------------------------------------------------------===//
+LogicalResult ExtFOp::verify() {
+  ChannelType srcType = getIn().getType();
+  ChannelType dstType = getOut().getType();
+
+  if (srcType.getDataBitWidth() < dstType.getDataBitWidth()) {
```

Comment:
Apologies I missed this in my first review but this should be `>`. If the bitwidths are equal that would of course be a no-op but that's still valid (a folder should fold this case away). Actually you can just define the function's body as `verifyExtOp(*this);`. The `verifyExtOp` function is defined in this file.